synapse/events/utils.py

```diff
@@ -79,13 +79,15 @@ def prune_event_dict(room_version: RoomVersion, event_dict: dict) -> dict:
         "state_key",
         "depth",
         "prev_events",
-        "prev_state",
         "auth_events",
         "origin",
         "origin_server_ts",
-        "membership",
     ]
 
+    # Room versions from before MSC 2176 had additional allowed keys.
```

Comment:
Extreme nitpick:
```suggestion
    # Room versions from before MSC2176 had additional allowed keys.
```

(applies throughout)